app/src/main/java/com/dessalines/thumbkey/keyboards/ESMessageEase.kt

```diff
@@ -640,13 +692,50 @@ val KB_ES_MESSAGEEASE_SHIFTED =
         ),
     )
 
+val NUM_KEY_2 = NUMERIC_KEYBOARD_MESSAGEASE.arr[0][1]
+val NUM_KEY_3 = NUMERIC_KEYBOARD_MESSAGEASE.arr[0][2]
+val KB_ES_MESSAGEEASE_NUM =
+    NUMERIC_KEYBOARD_MESSAGEASE.copy(
+        listOf(
+            listOf(
+                NUMERIC_KEYBOARD_MESSAGEASE.arr[0][0],
+                NUM_KEY_2.copy(
+                    center = NUM_KEY_2.center,
+                    swipes =
+                        NUM_KEY_2.swipes?.plus(
+                            SwipeDirection.RIGHT to
+                                KeyC(
+                                    display = KeyDisplay.TextDisplay("¡"),
+                                    action = KeyAction.CommitText("¡"),
+                                ),
+                        ),
+                ),
+                NUM_KEY_3.copy(
+                    center = NUM_KEY_3.center,
+                    swipes =
+                        NUM_KEY_3.swipes?.plus(
+                            SwipeDirection.LEFT to
+                                KeyC(
+                                    display = KeyDisplay.TextDisplay("¿"),
+                                    action = KeyAction.CommitText("¿"),
+                                ),
+                        ),
+                ),
+                NUMERIC_KEYBOARD_MESSAGEASE.arr[0][3],
+            ),
+            NUMERIC_KEYBOARD_MESSAGEASE.arr[1],
+            NUMERIC_KEYBOARD_MESSAGEASE.arr[2],
+            NUMERIC_KEYBOARD_MESSAGEASE.arr[3],
+        ),
+    )
+
 val KB_ES_MESSAGEEASE: KeyboardDefinition =
     KeyboardDefinition(
         title = "español messageease",
         modes =
             KeyboardDefinitionModes(
                 main = KB_ES_MESSAGEEASE_MAIN,
                 shifted = KB_ES_MESSAGEEASE_SHIFTED,
-                numeric = NUMERIC_KEYBOARD_MESSAGEASE,
+                numeric = KB_ES_MESSAGEEASE_NUM,
```

Comment:
Okay now its clear, you actually need multiple messageease number keyboards.

I'd just define the full ones separately instead of this array manipulation. Otherwise corrections and additions are going to be much more difficult for newcomers to add.

Reply:
Done, I've pushed a commit defining separate full numeric keyboards for english and spanish MessagEase layouts.

That being said, when #700 lands I'll update the spanish one with the newer functions.